### ISSUE TITLE
Performance Gap on my own training pipeline

### ISSUE THREAD

**Issue by GurrenLagann97** [GAIR-NLP/LIMO#9]
Hi, I use your data directly on my own training pipeline, but got remarkable performance gap：

![Image](https://github.com/user-attachments/assets/8ff7e2f1-2f71-4a96-8a59-eb387d983c33)
I wonder what training hyper parameter you use, such as learning rate and training epoch, etc.
Another question is that I use qwen 2.5 official eval pipeline, is it causes gap compare to your eval pipeline?
Thanks!

**Comment by GurrenLagann97** [comment]
By the way I tried model and eval code provided in the repo, but gap still exists on aime24 eval dataset, which is 33.3, I wonder what cause this gap because aime answers only have number.

**Comment by HuangZhen02** [comment]
Hi, you can find our detailed training configuration in: `https://github.com/GAIR-NLP/LIMO/blob/main/train/examples/train_limo.yaml`, we trained using 4 nodes with 32 GPUs in total (so the actual batch size is 32). 

Also, make sure your max_output_token when doing evaluation is enough (we use 32768 in our paper).

**Comment by BLeaves** [comment]
Thank you for your interest in our project and for raising this issue. It appears you may have evaluated LIMO using greedy decoding. For accurate comparison, please use our official hyperparameter configuration:

```bash
--temperature 0.6 \
--n_sampling 8 \
--max_tokens 32768 \
```

**Comment by lockylockylocky** [comment]
Hi, in your paper you mentioned you generate 16 samples with temperature setting of 0.7 for AIME and AMC, and greedy decoding with single sample for other datasets. It seems contradict with your official hyperparameter configuration mentioned above. So how should I set my hyperparameter?
